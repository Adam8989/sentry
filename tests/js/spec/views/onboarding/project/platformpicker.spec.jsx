import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import {Client} from 'app/api';
import PlatformPicker from 'app/views/onboarding/project/platformpicker';
import {categoryLists} from 'app/views/onboarding/utils';
import sinon from 'sinon';

describe('PlatformPicker', function() {
  beforeEach(function() {
    this.sandbox = sinon.sandbox.create();
    this.stubbedApiRequest = this.sandbox.stub(Client.prototype, 'request');
  });

  afterEach(function() {
    this.sandbox.restore();
  });

  describe('render()', function() {
    const baseProps = {
      location: {query: {}},
      params: {
        setPlatform: '',
        platform: '',
        isActive: ''
      }
    };

    it('should only render Mobile platforms under Mobile tab', function() {
      let props = {
        ...baseProps
      };

      let wrapper = shallow(<PlatformPicker {...props} />);
      expect(toJson(wrapper)).toMatchSnapshot();
      wrapper.setState({tab: 'Mobile'});
      let filteredPlatforms = wrapper
        .find('PlatformCard')
        .map(node => node.prop('platform'));

      expect(filteredPlatforms).not.toContain('java');
      expect(filteredPlatforms).toContain(categoryLists.Mobile[0]);
    });

    it('should render renderPlatformList with Python when filtered with P', function() {
      let props = {
        ...baseProps
      };

      let wrapper = shallow(<PlatformPicker {...props} />);
      expect(toJson(wrapper)).toMatchSnapshot();
      wrapper.setState({tab: 'All', filter: 'p'});
      let filteredPlatforms = wrapper
        .find('PlatformCard')
        .map(node => node.prop('platform'));
      expect(filteredPlatforms).not.toContain('java');
      expect(filteredPlatforms).toContain('python-flask');
    });

    it('should render renderPlatformList with community SDKs message if platform not found', function() {
      let props = {
        ...baseProps
      };

      let wrapper = shallow(<PlatformPicker {...props} />);
      expect(toJson(wrapper)).toMatchSnapshot();
      wrapper.setState({filter: 'aaaaaa'});
      expect(wrapper.text()).toContain('Not finding your platform?');
    });
  });
});
