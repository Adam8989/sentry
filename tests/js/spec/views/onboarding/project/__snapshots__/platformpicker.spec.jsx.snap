// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformPicker render() should only render Mobile platforms under Mobile tab 1`] = `
<div
  className="platform-picker"
>
  <ul
    className="nav nav-tabs"
  >
    <li
      style={
        Object {
          "float": "right",
          "marginRight": 0,
        }
      }
    >
      <div
        className="platform-filter-container"
      >
        <span
          className="icon icon-search"
        />
        <input
          className="platform-filter"
          label="Filter"
          onChange={[Function]}
          placeholder="Filter"
          type="text"
        />
      </div>
    </li>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      Popular
    </ListLink>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      Frontend
    </ListLink>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      Mobile
    </ListLink>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      Backend
    </ListLink>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      All
    </ListLink>
  </ul>
  <ul
    className="client-platform-list platform-tiles"
  >
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="java"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="javascript"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="node-express"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="php-laravel"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="php-symfony2"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="python-django"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="ruby-rails"
    />
  </ul>
</div>
`;

exports[`PlatformPicker render() should render renderPlatformList with Python when filtered with P 1`] = `
<div
  className="platform-picker"
>
  <ul
    className="nav nav-tabs"
  >
    <li
      style={
        Object {
          "float": "right",
          "marginRight": 0,
        }
      }
    >
      <div
        className="platform-filter-container"
      >
        <span
          className="icon icon-search"
        />
        <input
          className="platform-filter"
          label="Filter"
          onChange={[Function]}
          placeholder="Filter"
          type="text"
        />
      </div>
    </li>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      Popular
    </ListLink>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      Frontend
    </ListLink>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      Mobile
    </ListLink>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      Backend
    </ListLink>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      All
    </ListLink>
  </ul>
  <ul
    className="client-platform-list platform-tiles"
  >
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="java"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="javascript"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="node-express"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="php-laravel"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="php-symfony2"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="python-django"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="ruby-rails"
    />
  </ul>
</div>
`;

exports[`PlatformPicker render() should render renderPlatformList with community SDKs message if platform not found 1`] = `
<div
  className="platform-picker"
>
  <ul
    className="nav nav-tabs"
  >
    <li
      style={
        Object {
          "float": "right",
          "marginRight": 0,
        }
      }
    >
      <div
        className="platform-filter-container"
      >
        <span
          className="icon icon-search"
        />
        <input
          className="platform-filter"
          label="Filter"
          onChange={[Function]}
          placeholder="Filter"
          type="text"
        />
      </div>
    </li>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      Popular
    </ListLink>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      Frontend
    </ListLink>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      Mobile
    </ListLink>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      Backend
    </ListLink>
    <ListLink
      activeClassName="active"
      index={false}
      isActive={[Function]}
      onClick={[Function]}
      to=""
    >
      All
    </ListLink>
  </ul>
  <ul
    className="client-platform-list platform-tiles"
  >
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="java"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="javascript"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="node-express"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="php-laravel"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="php-symfony2"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="python-django"
    />
    <PlatformCard
      className=""
      onClick={[Function]}
      platform="ruby-rails"
    />
  </ul>
</div>
`;
